pkg/store/consul.go

```diff
@@ -91,11 +105,55 @@ func (s *Consul) Get(key string) (*KVPair, error) {
 }
 
 // Put a value at "key"
-func (s *Consul) Put(key string, value []byte) error {
-	p := &api.KVPair{Key: s.normalize(key), Value: value}
-	if s.client == nil {
-		log.Error("Error initializing client")
+func (s *Consul) Put(key string, value []byte, opts *WriteOptions) error {
+
+	key = s.normalize(key)
+
+	p := &api.KVPair{
+		Key:   key,
+		Value: value,
+	}
+
+	if opts != nil && opts.Ephemeral {
+		if _, ok := s.sessions[key]; !ok {
+			entry := &api.SessionEntry{
+				Behavior: api.SessionBehaviorDelete,
+				TTL:      s.ephemeralTTL.String(),
+			}
+
+			// Create global ephemeral keys session
+			session, _, err := s.client.Session().Create(entry, nil)
+			if err != nil {
+				return err
+			}
+
+			// Create lock option with the
+			// EphemeralSession
+			lockOpts := &api.LockOptions{
+				Key:     key,
+				Session: session,
+			}
+
+			// Lock and ignore if lock is held
+			// It's just a placeholder for the
+			// ephemeral behavior
+			lock, _ := s.client.LockOpts(lockOpts)
```

Comment:
We need a lock as the holder of the session with `Behavior: Delete` to mimic Zookeeper's ephemeral node behavior :)